doc/src/manual/calling-c-and-fortran-code.md

```diff
@@ -854,18 +854,135 @@ it must be handled in other ways.
 
 In some cases, the exact name or path of the needed library is not known in
 advance and must be computed at run time. To handle such cases, the library
-component specification can be a value such as `Libdl.LazyLibrary`. For
-example, in `@ccall blas.dgemm()`, there can be a global defined as `const blas
-= LazyLibrary("libblas")`. The runtime will call `dlsym(:dgemm, dlopen(blas))`
-when the `@ccall` itself is executed. The `Libdl.dlopen` function can be
-overloaded for custom types to provide alternate behaviors. However, it is
-assumed that the library location does not change once it is determined, so the
-result of the call can be cached and reused. Therefore, the number of times the
-expression executes is unspecified, and returning different values for multiple
-calls results in unspecified behavior.
-
-If even more flexibility is needed, it is possible
-to use computed values as function names by staging through [`eval`](@ref) as follows:
+component specification can be a value such as `Libdl.LazyLibrary`. The runtime
+will call `Libdl.dlopen` on that object when first used by a `ccall`.
+
+### [Using LazyLibrary for Lazy Loading](@id man-lazylibrary)
+
+[`Libdl.LazyLibrary`](@ref) provides a thread-safe mechanism for deferring library loading
+until first use. This is the recommended approach for library initialization in modern Julia code.
+
+A `LazyLibrary` represents a library that opens itself (and its dependencies) automatically
+on first use in a `ccall()`, `@ccall`, `dlopen()`, `dlsym()`, `dlpath()`, or `cglobal()`.
+The library is loaded exactly once in a thread-safe manner, and subsequent calls reuse the
+loaded library handle.
+
+#### Basic Usage
+
+```julia
+using Libdl
+
+# Define a LazyLibrary as a const for optimal performance
+const libz = LazyLibrary("libz")
+
+# Use directly in @ccall - library loads automatically on first call
+@ccall libz.deflate(strm::Ptr{Cvoid}, flush::Cint)::Cint
+
+# Also works with ccall
+ccall((:inflate, libz), Cint, (Ptr{Cvoid}, Cint), strm, flush)
+```
+
+#### Platform-Specific Libraries
+
+For code that needs to work across different platforms:
+
+```julia
+const mylib = LazyLibrary(
+    if Sys.iswindows()
+        "mylib.dll"
+    elseif Sys.isapple()
+        "libmylib.dylib"
+    else
+        "libmylib.so"
+    end
+)
+```
+
+#### Libraries with Dependencies
+
+When a library depends on other libraries, specify the dependencies to ensure
+they load in the correct order:
+
+```julia
+const libfoo = LazyLibrary("libfoo")
+const libbar = LazyLibrary("libbar"; dependencies=[libfoo])
+
+# When libbar is first used, libfoo is loaded first automatically
+@ccall libbar.bar_function(x::Cint)::Cint
+```
+
+#### Lazy Path Construction
+
+For libraries whose paths are determined at runtime, use `LazyLibraryPath`:
```

Comment:
Maybe make a small note as to why you didn't need to use `LazyLibraryPath` when doing the OS-specific paths above.